nodebuilder/tests/p2p_test.go

```diff
@@ -86,131 +95,123 @@ Steps:
 	err := bridge.Start(ctx)
 	require.NoError(t, err)
-	bridgeAddr := host.InfoFromHost(bridge.Host)
 
+	// use BN as the bootstrapper
+	bootstrapper := host.InfoFromHost(bridge.Host)
+
+	// create FN with BN as bootstrapper
 	cfg = nodebuilder.DefaultConfig(node.Full)
 	setTimeInterval(cfg, defaultTimeInterval)
 	full := sw.NewNodeWithConfig(
 		node.Full,
 		cfg,
-		nodebuilder.WithBootstrappers([]peer.AddrInfo{*bridgeAddr}),
+		nodebuilder.WithBootstrappers([]peer.AddrInfo{*bootstrapper}),
 	)
 
+	// create LN with BN as bootstrapper
 	cfg = nodebuilder.DefaultConfig(node.Light)
 	setTimeInterval(cfg, defaultTimeInterval)
-	cfg.P2P.PeerExchange = true
 	light := sw.NewNodeWithConfig(
 		node.Light,
 		cfg,
-		nodebuilder.WithBootstrappers([]peer.AddrInfo{*bridgeAddr}),
+		nodebuilder.WithBootstrappers([]peer.AddrInfo{*bootstrapper}),
 	)
+
+	// start FN and LN and ensure they are both connected to BN as a bootstrapper
 	nodes := []*nodebuilder.Node{full, light}
-	ch := make(chan struct{})
-	sub, err := light.Host.EventBus().Subscribe(&event.EvtPeerConnectednessChanged{})
-	require.NoError(t, err)
-	defer sub.Close()
 	for index := range nodes {
 		require.NoError(t, nodes[index].Start(ctx))
-		assert.Equal(t, *bridgeAddr, nodes[index].Bootstrappers[0])
-		assert.True(t, nodes[index].Host.Network().Connectedness(bridgeAddr.ID) == network.Connected)
+		assert.Equal(t, *bootstrapper, nodes[index].Bootstrappers[0])
+		assert.True(t, nodes[index].Host.Network().Connectedness(bootstrapper.ID) == network.Connected)
 	}
-	addrFull := host.InfoFromHost(full.Host)
-	go func() {
-		for e := range sub.Out() {
-			connStatus := e.(event.EvtPeerConnectednessChanged)
-			if connStatus.Peer == full.Host.ID() && connStatus.Connectedness == network.NotConnected {
-				ch <- struct{}{}
-			}
+
+	for {
+		if ctx.Err() != nil {
+			t.Fatal(ctx.Err())
+		}
+		if light.Host.Network().Connectedness(host.InfoFromHost(full.Host).ID) == network.Connected {
+			// LN discovered FN successfully and is now connected
+			break
 		}
-	}()
-
-	// ensure that the light node is connected to the full node
-	assert.True(t, light.Host.Network().Connectedness(addrFull.ID) == network.Connected)
-
-	sw.Disconnect(t, light.Host.ID(), full.Host.ID())
-	require.NoError(t, full.Stop(ctx))
-	select {
-	case <-ctx.Done():
-		t.Fatal("peer was not disconnected")
-	case <-ch:
-		assert.True(t, light.Host.Network().Connectedness(addrFull.ID) == network.NotConnected)
 	}
 }
 
 /*
-Test-Case: Restart full node discovery after one node is disconnected
+Test-Case: Full node discovery of disconnected full nodes
 Steps:
 1. Create a Bridge Node(BN)
 2. Start a BN
 3. Create 2 full nodes with bridge node as bootstrapper peer and start them
 4. Check that nodes are connected to each other
-5. Create one more node with disabled discovery
-6. Disconnect FNs from each other
-7. Check that the last FN is connected to one of the nodes
+5. Disconnect the FNs
+6. Create one more node with disabled discovery
```

Comment:
After reading description I was a bit confused how node could be discovered, when it has discovery disabled. Turned out it description is a bit misleading. According to code last node is not really have disabled discovery, but rather it is not searching for other nodes, but still advertises itself to allow it to be discovered by others.

Reply:
added description there